pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -827,6 +835,69 @@ allChainsLabel:
 	return mutable.Listener
 }
 
+func buildSidecarListenerTLSContext(tlsOpts *networking.Server_TLSOptions) *auth.DownstreamTlsContext {
```

Comment:
cant we use the function in the gateway?"

Reply:
I have thought about that, but the tls mode supported is not same

Reply:
Mode seems different, but part of logic to build the tls contexts can be reused but not sure if it is worth the refactor I guess.